pkg/kwokctl/runtime/util.go

```diff
@@ -116,10 +118,30 @@ func applyComponentPatch(component *internalversion.Component, patch internalver
 
 	component.Volumes = append(component.Volumes, patch.ExtraVolumes...)
 	component.Envs = append(component.Envs, patch.ExtraEnvs...)
-
 	for _, a := range patch.ExtraArgs {
-		component.Args = append(component.Args, fmt.Sprintf("--%s=%s", a.Key, a.Value))
+		if a.Override {
+			component.Args = applyComponentArgsOverride(context.Background(), component.Args, a)
```

Comment:
Don't use context.Background(), let it be passed by the caller step by step

Reply:
np, This was my first thought too, but I don't know your preference and since I saw some context.Background() in the project, I just tried it.
Now I updated.